src/application_settings/container_base.py

```diff
@@ -0,0 +1,226 @@
+    def get(cls: type[_ContainerT], reload: bool = False) -> _ContainerT:
+        """Access method for the singleton."""
+
+        if (_the_container_or_none := cls._get()) is None or reload:
+            # no config has been made yet or it needs to be reloaded,
+            # so let's instantiate one and keep it in the global store
+            _the_config = cls._create_instance()
+            _the_config._set()
+        else:
+            _the_config = _the_container_or_none
+        return _the_config
+
+    def update(self: _ContainerT, changes: dict[str, dict[str, Any]]) -> _ContainerT:
+        "Update and save the settings with data specified in changes; not meant for config"
+        # filter out fields that are both in changes and an attribute of the SettingsContainer
+        _sections_to_update = {
+            fld for fld in fields(self) if fld.init and fld.name in changes.keys()
+        }
+
+        # update the sections and keep them in a dict
+        # actually sections: dict[str, _ContainerSectionT]
+        # but MyPy doesn't swallow that
+        updated_sections: dict[str, Any] = {
+            fld.name: _update_section(getattr(self, fld.name), changes[fld.name])
+            for fld in _sections_to_update
+        }
+        new_settings = replace(self, **updated_sections)
+        new_settings._set()  # pylint: disable=protected-access
+        new_settings._save()  # pylint: disable=protected-access
+        return new_settings
+
+    @classmethod
+    def _create_instance(cls: type[_ContainerT]) -> _ContainerT:
+        """Instantiate the Container."""
+
+        # get whatever is stored in the config/settings file
+        data_stored = cls._get_stored_data()
+        # filter out fields that are both stored and an attribute of the Container
+        _data_fields = {
+            fld for fld in fields(cls) if fld.init and fld.name in data_stored.keys()
+        }
+        # instantiate the sections and keep them in a dict
+        # actually sections: dict[str, _ContainerSectionT]
+        # but MyPy doesn't swallow that
+        sections: dict[str, Any] = {
+            fld.name: cls._instantiate_section(fld.type, data_stored[fld.name])
+            for fld in _data_fields
+        }
+
+        # instantiate the Container with the sections
+        return cls(**sections)
+
+    @classmethod
+    def _get_stored_data(cls) -> dict[str, Any]:
+        """Get the data stored in the toml file"""
+        data_stored: dict[str, Any] = {}
+
+        if path := cls.filepath():
+            if (ext := path.suffix[1:]) == str(FileFormat.TOML.value).lower():
+                with path.open(mode="rb") as fptr:
+                    data_stored = tomllib.load(fptr)
+            elif ext == str(FileFormat.JSON.value):
+                with path.open(mode="r") as fptr:
+                    data_stored = json.load(fptr)
+            else:
+                print(f"Unknown file format {path.suffix[1:]} given in {path}.")
```

Comment:
I think you misunderstood what I meant. In the printing you're using `path.suffix[1:]`, but we could use the `ext` variable defined at line 164 :-).

Reply:
Right, now I get it :D